### DISCUSSION TITLE
How to use IDP initiated login

### DISCUSSION THREAD

**Discussion by ankush-maherwal** [keycloak/keycloak#21540] [Q&A]
I am setup in which there are four systems 

1) React App 1 (This App uses Keycloak as IDP)
2) Keycloak (Used as IDP)
3) React app 2 (SP which uses Keycloak as IDP)
4) React app 3 (SP which uses Keycloak as IDP)

In my case users will login to React App 1 using Keycloak and once logged then users will see the link to login to React app 2 and React app 3 (This will be SAML based SSO login from React App1 into React app 2 and React app 3)

I will load React app 2 and React app 3 internally inside React app 1

How I can implement SAML based SSO login to React app 2 and React app 3 from React app 1

Any advice is highly appreciated

**Comment by sohaibaumer** [comment]
In Keycloak, create a realm for each React app. Then, create a client for each React app. In the client settings, configure the SAML settings. 
In each React app, you will need to configure the SAML settings like IDP metadata URL, the SP metadata URL, the assertion consumer service URL etc in the config.js file. 
Implement the SAML login flow in React app 1 like:
- The user clicks on the "Login to React app 2" or "Login to React app 3" link.
- React app 1 redirects the user to the IDP login page.
- The user logs in to the IDP.
- The IDP redirects the user back to React app 1.
- React app 1 receives the SAML response from the IDP.
- React app 1 validates the SAML response.
- React app 1 logs the user in to React app 2 or React app 3.
